sridha2412/assignment2/q1.py

```diff
@@ -2,29 +2,41 @@
 
 # node class`
 class Node:
-    # constuctor to initlalize a node object
-    def __init__(self, data):
+    def __init__(self, data: int):
+        """ constuctor to initlalize a node object """
         self.data = data
         self.left = None
         self.right = None
 
-# if key is present, print ancestors else return false
-def print_ancestors(root, key):
-    # base case
+def find_ancestors(root: Node, key: int, ancestors: list) -> bool:
+    """ if key is present, return list of ancestors else return false """
+    # base cases
     if root is None:
         return False
-
     if root.data == key:
         return True
 
+    # store all nodes in list
+    ancestors.append(root.data)
+
     # search for key in left and right subtrees
-    if print_ancestors(root.left, key) or print_ancestors(root.right, key) :
-        print (root.data, end = " ")
-        return True
+    if find_ancestors(root.left, key, ancestors) or find_ancestors(root.right, key, ancestors) :
+        return ancestors
+
+    # if not ancestor, delete from list
+    ancestors.pop()
 
     return False
 
 
+#       7
+#      / \
+#     3   4
+#    / \   \
+#   2   5   8
+#  / \
+# 1   6
+
```

Comment:
Please remove the code that executes on import, you can move it into a unit test.